File: SAV_Frontend/src/Models/Intervention.ts
import { Piece } from './Piece';
import { Reclamation } from './Reclamation';
import { ResponsableSAV } from './ResponsableSAV';

export interface Intervention {
  id: number;
  dateIntervention?: Date;
  montantFacture?: number;
  estGratuit?: boolean;
  reclamationId: number;

  reclamation?: Reclamation;
  responsableSAVId: number;

  responsableSAV?: ResponsableSAV;
  pieces?: Piece[];
}

File: SAV_Backend/SAV_Backend/Models/Intervention.cs
using System.IO.Pipelines;
using System.Text.Json.Serialization;

namespace SAV_Backend.Models
{
    public class Intervention
    {
        public int Id { get; set; }
        public DateTime DateIntervention { get; set; }
        public double? MontantFacture { get; set; }
        public bool? EstGratuit { get; set; }
        public int? ReclamationId { get; set; }

        [JsonIgnore]
        public Reclamation? Reclamation { get; set; }
   
        public int ResponsableSAVId { get; set; }
        [JsonIgnore]
        public ResponsableSAV? ResponsableSAV { get; set; }

        public ICollection<Piece>? Pieces { get; set; }

    }



}
